### PR TITLE
Navigation Updates

### PR DESCRIPTION
**Pull request by martypowell**
- Make hovering the nav smoother
- Ensure search works with the nav

I know this is kinda vague but we changed a lot.

Maybe it's worth creating some requirements?

FYI @danfox01

### CODE REVIEW

File: js/navigation-search.js [hunk 1]
```diff
@@ -51,25 +53,41 @@ bcpl.navigationSearch = (($) => {
 		}
 	};
 
+	const onDocumentClick = (clickEvent) => {
+		const $target = $(clickEvent.target);
+		const isTargetSearchButtonContainer = $target.closest('.search-button-container').length;
```
Comment (mxsnyder) [inline js/navigation-search.js]: Add a variable for the selector

File: js/site-search.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		if (!doesSearchHaveValue) {
+			$elmToHide = $(siteSearchClearIconSelector);
+			$elmToShow = $(siteSearchSearchIconSelector);
+		}
+
+		$elmToHide.hide();
+		$elmToShow.show();
+	};
+
+	const init = () => {
```
Comment (mxsnyder) [inline js/site-search.js]: You don't event need these in an `init()` function.

File: stylesheets/partials/features/_header-main-menu.scss [hunk 3]
```diff
@@ -29,6 +29,11 @@
 		font-size: 1.5em;
 	}
 	.nav-and-search {
+		
+		button:focus {
+			outline-color: transparent;	
+		}
```
Comment (mxsnyder) [inline stylesheets/partials/features/_header-main-menu.scss]: You should talk to Dan and Sam about this if you haven't yet. We'll need some sort of focus indication for keyboard-only users.

Comment (martypowell) [reply stylesheets/partials/features/_header-main-menu.scss]: ok, ill talk to dan before merging

File: js/site-search.js (hunk 2 above)
Comment (martypowell) [reply js/site-search.js]: good point, i was lazy and just copied and pasted

File: js/navigation-search.js (hunk 1 above)
Comment (martypowell) [reply js/navigation-search.js]: good point, updated

File: stylesheets/partials/features/_header-main-menu.scss (hunk 3 above)
Comment (martypowell) [reply stylesheets/partials/features/_header-main-menu.scss]: added back, looks good confirmed by @danfox01

**Pull request merged**